sysml.library/Systems Library/Items.sysml

```diff
@@ -29,29 +35,53 @@ package Items {
 		item shape : Item :>> spaceBoundary;
 		
 		/**
-		 * Other shapes that spatially envelop this Item.
+		 * Each enveloping shape is the shape of an Item that includes this Item in space and time.
 		 */
-		item envelopingShapes : Item[0..*] {
-			attribute :>> innerSpaceDimension = shape.innerSpaceDimension;
+		item envelopingShapes : Item[0..*] {  /* envelopingBoundaries?	Occurrence::enclosed => included, envelopingBoundaries -> enclose ?*/
+
+			  /* Enables 2 dim items to be enveloped by 2 or 3 dim shapes. */
+			attribute :>> innerSpaceDimension = (Item::innerSpaceDimension == 3  | Item::outerSpaceDimension == 3
+							     ? 2 : Item::outerSpaceDimension-1);   /* Curves bounding flat surfaces SHOULD BE COPLANAR ? */
+			assert constraint { Item::innerSpaceDimension < 3 implies notEmpty(outerSpaceDimension) }
```

Comment:
`/* envelopingBoundaries?	Occurrence::enclosed => included, envelopingBoundaries -> enclose ?*/`
`/* Curves bounding flat surfaces SHOULD BE COPLANAR ? */`

These questions need to be resolved or the comments removed.

Reply:
Was thinking we might want to have fewer "en" words (enclose, envelop), maybe by using them only in SysML and changing "enclosed" to "included" in Kernel (eg, timeIncludedOccurrences).  Then envelopingShapes would be enclosingShapes.

Reply:
Not worth worrying more about naming now. Time to move on.